src/status_im/events.cljs

```diff
@@ -1,369 +1,372 @@
+
+;;TODO :replace by named events
+(fx/defn set-once-event
+  {:events [:set-once]}
+  [{:keys [db]} k v]
+  (when-not (get db k)
+    {:db (assoc db k v)}))
+
+;;TODO :replace by named events
+(fx/defn set-in-event
+  {:events [:set-in]}
+  [{:keys [db]} path v]
+  {:db (assoc-in db path v)})
+
+(defn on-ramp<-rpc [on-ramp]
+  (clojure.set/rename-keys on-ramp {:logoUrl :logo-url
+                                    :siteUrl :site-url}))
+
+(fx/defn crypto-loaded-event
+  {:events [::crypto-loaded]}
+  [{:keys [db]} on-ramps]
+  {:db (assoc
+        db
+        :buy-crypto/on-ramps
+        (map on-ramp<-rpc on-ramps))})
+
+(fx/defn buy-crypto-ui-loaded
+  {:events [:buy-crypto.ui/loaded]}
+  [_]
+  {::json-rpc/call [{:method     "wallet_getCryptoOnRamps"
+                     :params     []
+                     :on-success (fn [on-ramps]
+                                   (re-frame/dispatch [::crypto-loaded on-ramps]))}]})
+
+(fx/defn open-buy-crypto-screen
+  {:events [:buy-crypto.ui/open-screen]}
+  [cofx]
+  (fx/merge
+   cofx
+   (navigation/open-modal :buy-crypto nil)
+   (wallet/keep-watching-history)))
+
+;; Information Box
+
+(def closable-information-boxes
+  "[{:id      information box id
+     :global? true/false (close information box across all profiles)}]"
+  [])
+
+(defn information-box-id-hash [id public-key global?]
+  (if global?
+    (hash id)
+    (hash (str public-key id))))
+
+(fx/defn close-information-box
+  {:events [:close-information-box]}
+  [{:keys [db]} id global?]
+  (let [public-key (get-in db [:multiaccount :public-key])
+        hash       (information-box-id-hash id public-key global?)]
+    {::async-storage/set! {hash true}
+     :db (assoc-in db [:information-box-states id] true)}))
+
+(fx/defn information-box-states-loaded
+  {:events [:information-box-states-loaded]}
+  [{:keys [db]} hashes states]
+  {:db (assoc db :information-box-states (reduce
+                                          (fn [acc [id hash]]
+                                            (assoc acc id (get states hash)))
+                                          {} hashes))})
+
+(fx/defn load-information-box-states
+  {:events [:load-information-box-states]}
+  [{:keys [db]}]
+  (let [public-key            (get-in db [:multiaccount :public-key])
+        {:keys [keys hashes]} (reduce (fn [acc {:keys [id global?]}]
+                                        (let [hash (information-box-id-hash
+                                                    id public-key global?)]
+                                          (-> acc
+                                              (assoc-in [:hashes id] hash)
+                                              (update :keys #(conj % hash)))))
+                                      {} closable-information-boxes)]
+    {::async-storage/get {:keys keys
+                          :cb   #(re-frame/dispatch
+                                  [:information-box-states-loaded hashes %])}}))

```

Comment:
Can we please make changes in the file prettier .. There's lots of unnecessary changes detected by GitHub